mlflow/utils/validation.py

```diff
@@ -163,6 +163,8 @@ def _validate_metric(key, value, timestamp, step):
             INVALID_PARAMETER_VALUE,
         )
 
+    _validate_length_limit("Metric name", MAX_ENTITY_KEY_LENGTH, key)
```

Comment:
This would break code that logs metrics with key exceeding `MAX_ENTITY_KEY_LENGTH` (250).